super-cool-game/Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerMovement : MonoBehaviour {
+    /* Handles the movement of the player.
+     * This script is always attached to the player's character game object,
+     * while PlayerInput is attached only when the character's owner is the local player.
+     */
+    
+    // These ones can be edited in the inspector
+    public float speed = 3.0f;
+    public float maxSpeed = 9.0f;
+    public float jumpHeight = 6.0f;
+    public float checkGroundDistance = 1.0f;
+    public LayerMask groundLayer;
+
+    private Rigidbody2D rb;
+    private float moveX;
+
+    // Use this for initialization
+    void Start() {
+        // Get Rigidbody from the GameObject
+        rb = gameObject.GetComponent<Rigidbody2D>();
+    }
+
+    // Fixed update is for Physics-based update
+    void FixedUpdate () {
+        ComputeMovement();
+    }
+
+    void ComputeMovement () {
+        
+
+        // Resultant of movement
+        rb.velocity = new Vector2(moveX, rb.velocity.y);
```

Comment:
This could cause a race condition with PlayerInput.Update. If this gets executed before PlayerInput.Update, moveX has never been assigned. Make sure to initialize it.